src/export/Export.h

```diff
@@ -108,6 +108,17 @@ class AUDACITY_DLL_API ExportPlugin /* not final */
     * of channels in exported file. -1 for unspecified */
    virtual int SetNumExportChannels() { return -1; }
 
+   // TODO:  the next three virtuals might take arguments, as does
+   // GetFormat()
+
+   //! May modify global settings affecting export;
+   //! not responsible to revert changes
+   virtual void SetupUploadFormat();
```

Comment:
I strongly disagree with this change. `ExportPlugin` has nothing to do with uploading.